lib/utilities/firebase_service.dart

```diff
@@ -65,4 +76,110 @@ class FirebaseService {
       return 'Unknown';
     }
   }
+
+  Future<void> unregisterFromEvent(String eventId, String userId) async {
+    try {
+      // Reference to the event document
+      final eventDoc = _firestore.collection('events-collection').doc(eventId);
+
+      // Update the registeredUsers array and increment slotsAvailable
+      await eventDoc.update({
+        'registeredUsers': FieldValue.arrayRemove([userId]),
+        'slotsAvailable': FieldValue.increment(1), // Increment available slots
+      });
+
+      // Query and delete the user's registration entry from the registration collection
+      final registrationQuery = await _firestore
+          .collection('registration')
+          .where('eventId', isEqualTo: eventId)
+          .where('userId', isEqualTo: userId)
+          .get();
+
+      for (var doc in registrationQuery.docs) {
+        await doc.reference.delete(); // Delete registration document
+      }
+
+      print('User $userId successfully unregistered from event $eventId');
+    } catch (e) {
+      print('Error unregistering from event: $e');
+    }
+  }
+  Future<List<SportEvent>> getUserRegisteredEvents(String userId, bool showFutureEvents) async {
+  try {
+    // Query the registration collection to fetch event IDs for the user
+    final registrationQuery = await _firestore
+        .collection('registration')
+        .where('userId', isEqualTo: userId)
+        .get();
+
+    final eventIds = registrationQuery.docs.map((doc) {
+      final data = doc.data() as Map<String, dynamic>;
+      return data['eventId'] as String;
+    }).toList();
+
+    // Ensure eventIds is not empty
+    if (eventIds.isEmpty) return [];
+
+    // Fetch all events from the collection and filter manually
+    final eventsSnapshot = await _firestore.collection('events-collection').get();
+
+    final filteredEvents = eventsSnapshot.docs.where((doc) {
+      final data = doc.data() as Map<String, dynamic>;
+      final eventDateTime = data['dateTime'] as Timestamp;
+      final isFutureEvent = eventDateTime.toDate().isAfter(DateTime.now());
+      return eventIds.contains(doc.id) && (showFutureEvents ? isFutureEvent : !isFutureEvent);
+    }).toList();
```

Comment:
Add the dateTime filter in the query to the database, do not load all into memory and then filter them in the app. Let the database do that. 
```dart
      final snapshot = await _firestore
          .collection('events-collection')
          .where('sportId', isEqualTo: sportId)
          .where('dateTime',
              isGreaterThan: Timestamp.now()) // Filter future events
          .get();
```

Reply:
used this code ,it ran fine and able to register , but getting error while fetching registered games in my games page , I looked in internet and came to know using "where" allows to filter for only 10 datas per each id